src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -2955,7 +2965,12 @@ protected function asDateTime($value)
         // Finally, we will just assume this date is in the format used by default on
         // the database connection and use that format to create the Carbon object
         // that is returned back out to the developers after we convert it here.
-        return Carbon::createFromFormat($this->getDateFormat(), $value);
+        if (static::$timezone) {
+            return Carbon::createFromFormat($this->getDateFormat(), $value, static::$timezone)
+                ->setTimezone(date_default_timezone_get());
```

Comment:
This takes the date from the database, which is stored in (for example) UTC format, and creates a `DateTimeInterface` or `Carbon` object that is in the PHP default timezone. In a Laravel app your PHP default timezone is `Config::get('app.timezone')`.

In other words, you've specified what timezone the database is using, but your app may not be using that timezone. You would expect the ORM to handle this detail for you, returning a DateTime object in the timezone of your app. This DateTime object is guaranteed to be equivalent to the date/time stored in the database, no matter what timezone your app is using. And of course, anything with `DateTimeInterface` can losslessly be converted to another timezone if desired.